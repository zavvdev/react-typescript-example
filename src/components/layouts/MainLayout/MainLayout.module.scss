.root {
  padding: 0 5% 200px 5%;

  .content {
    padding-top: 42px;
  }
}