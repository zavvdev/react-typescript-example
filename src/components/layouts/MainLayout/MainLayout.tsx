import Header from 'src/components/Header/Header';
import classes from 'src/components/layouts/MainLayout/MainLayout.module.scss';
import { MainLayoutProps } from 'src/components/layouts/MainLayout//MainLayout.types';

const MainLayout = (props: MainLayoutProps) => {
  const { children } = props;

  return (
    <div className={classes.root}>
      <Header />
      <div className={classes.content}>
        {children}
      </div>
    </div>
  );
};

export default MainLayout;
