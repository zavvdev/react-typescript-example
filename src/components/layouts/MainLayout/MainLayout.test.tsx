import { screen } from '@testing-library/react';
import render from 'src/tests/utils/render';
import MainLayout from 'src/components/layouts/MainLayout/MainLayout';

const textMock = 'textMock';

describe('<MainLayout />', () => {
  test('should render MainLayout component (snapshot)', () => {
    const { asFragment } = render(
      <MainLayout>
        <>{textMock}</>
      </MainLayout>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('should have child component in layout', () => {
    render(
      <MainLayout>
        <>{textMock}</>
      </MainLayout>,
    );
    expect(screen.getByText(textMock)).toBeInTheDocument();
  });
});
