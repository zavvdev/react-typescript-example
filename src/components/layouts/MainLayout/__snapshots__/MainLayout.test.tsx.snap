// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainLayout /> should render MainLayout component (snapshot) 1`] = `
<DocumentFragment>
  <div
    class="root"
  >
    <header
      class="root"
    >
      <img
        alt="appName"
        class="logo"
        src="https://test-file-stub"
      />
      <div
        class="root"
      >
        <div
          class="langWrap"
        >
          <button
            class="lang active"
            type="button"
          >
            en
          </button>
          <span>
            |
          </span>
        </div>
        <div
          class="langWrap"
        >
          <button
            class="lang"
            type="button"
          >
            ru
          </button>
        </div>
      </div>
    </header>
    <div
      class="content"
    >
      textMock
    </div>
  </div>
</DocumentFragment>
`;
